### PR TITLE
Miscellaneous Python documentation fixes

### LINKED ISSUE #1078: Cantera documentation: Python member functions erroneously labeled as properties
Looking at the current documentation, there are at least some instances where *member functions* are erroneously labeled as *properties*. As an example, `thermo.pyx` defines
```
def set_equivalence_ratio(self, phi, fuel, oxidizer, basis='mole'):
```
which is rendered as [property set_equivalence_ratio](https://cantera.org/documentation/docs-2.5/sphinx/html/cython/importing.html#cantera.Quantity.set_equivalence_ratio).

Beyond, there are some formatting glitches, e.g. for [property get_equivalence_ratio](https://cantera.org/documentation/docs-2.5/sphinx/html/cython/importing.html#cantera.Quantity.get_equivalence_ratio).

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

- Fix confusing labeling of all attributes of `SolutionArray` and `Quantity` class being labeled as `property` in the Sphinx docs
- Make extended arguments (other than input file, phase name, and adjacent phases list) to `Solution` keyword-only (I'm not sure they would have worked as positional arguments in the first place)
- Make extended arguments (other than reactants, products, and rate) to `Reaction` classes keyword-only (again, I think this was implicitly required in practice
- Fix a bunch of formatting issues in the Sphinx documentation, mostly with how references to function arguments are formatted. Using double backticks gives normal monospace formatting.
- Eliminate the remnants of the long-ago deprecated `WallSurface` class.

**If applicable, fill in the issue number this pull request is fixing**

Fixes #1078

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Review by ischoegl — DISMISSED** [review]
:tada: ... thanks for this fix @speth! While I'm not exactly sure what fixed the issue, I assume that it has to do with the signatures in the rst files? I verified with the GH artifact that things work. Also, thanks for making the formatting more consistent. Overall, this looks good to go from my perspective.

**Comment by speth** [comment]
The trick to fixing the "property" labels was realizing that they actually _were_ `property` objects, and this wasn't a weird Sphinx bug. These were just properties that returned the corresponding method of the underlying `Solution` object, but there's no way for Sphinx to know that. So the solution was instead to to wrap the methods that need to be forwarded without modification in a trivial function and copy the docstring from the parent function.

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]: Should these be double backticks as well?

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
        to be matched to the `SolutionArray`.
```

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
        keys correspond to `SolutionArray` attributes to be exported.
```

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
            include any arrays which were specified as ``extra`` variables when
```

Comment (bryanwweber) [reply interfaces/cython/cantera/composite.py]: Likewise on the next two lines.

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
        Write data specified by ``cols`` to an HDF container file named ``filename``.
```

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
        saved in the form of HDF attributes. An additional intermediate layer may
```

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
        Arguments ``compression`` and ``compression_opts`` are mapped to parameters
```

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]: Does this create link to this method?
```suggestion
        for `h5py.create_dataset`; in both cases, the choices of ``None`` results
```

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [inline interfaces/cython/cantera/composite.py]:
```suggestion
        Additional arguments (that is, ``*args`` and ``**kwargs``) are passed on to
```

File: interfaces/cython/cantera/kinetics.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/kinetics.pyx]:
```suggestion
        mechanism if ``indices`` is unspecified, or the equations for each
```

File: interfaces/cython/cantera/kinetics.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/kinetics.pyx]:
```suggestion
            definitions (see Eq. 9.75 in Kee, Coltrin, and Glarborg, *Chemically
```

File: interfaces/cython/cantera/kinetics.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/kinetics.pyx:425]: As an aside, we should consider reproducing this equation somewhere in the website documentation... presumably as a result of resolving https://github.com/Cantera/cantera/issues/1165

File: interfaces/cython/cantera/kinetics.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/kinetics.pyx]:
```suggestion
        `net_production_rates` property to get the net production rates for
```

File: interfaces/cython/cantera/onedim.py
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.py:360]:
```
Get the solution for the specified ``component`` at one ``point``, or for the full flame domain if ``point`` is not specified.
```

File: interfaces/cython/cantera/onedim.py
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.py]:
```suggestion
        Set the state of the the `Solution` object used for calculations
        to the temperature and composition at the point with index
```

File: interfaces/cython/cantera/onedim.py
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.py]:
```suggestion
        Additional arguments (that is, ``*args`` and ``**kwargs``) are passed on to
```

File: interfaces/cython/cantera/onedim.py
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.py]:
```suggestion
        Bilger mixture fraction by setting ``m="Bilger"``
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        component names as keywords and (lower bound, upper bound) tuples as
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        component names as keywords and (relative tolerance, absolute tolerance) tuples as the values.
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]: Same as above

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        Set an interrupt function to be called each time that :ct:`OneDim::eval` is
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        called. The signature of ``f`` is ``float f(float)``. The default
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        interrupt function is used to trap `KeyboardInterrupt` exceptions so
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        that ``ctrl-c`` can be used to break out of the C++ solver loop.
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        The signature of ``f`` is ``float f(float)``. The argument passed to ``f`` is
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        solve, before regridding. The signature of ``f`` is ``float f(float)``. The
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]: Is the argument the string `"0"`? Or is it an integer or float?

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        Return the underlying data specifying a ``domain``. This method is used as
```

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]: I don't understand the `, respectively` in this sentence.

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        (for example, ``inlet``, ``outlet``, etc.), can be specified either by name or the
```

But I'm not sure this parenthetical actually clarifies anything, so it could probably be removed.

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        Restore a ``domain`` from underlying data. This method is used as
```

To what "underlying data" does this refer? Underlying what, from where?

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        Derived classes set default values for ``domain`` and ``other``, where
```

Same comment about `respectively` and the `e.g. inlet, outlet` parenthetical.

File: interfaces/cython/cantera/onedim.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/onedim.pyx]:
```suggestion
        Set the minimum grid spacing on ``domain``. If ``domain`` is ``None``, then
```

File: interfaces/cython/cantera/reaction.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reaction.pyx:200]: Should A, b, and E_a (note it's E in the equation) be in math mode here?

File: interfaces/cython/cantera/reaction.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reaction.pyx:1205]: Does this not link to the corresponding property of the `Reaction` class?

Comment (bryanwweber) [reply interfaces/cython/cantera/reaction.pyx:1205]: This same comment applies to the Chebyshev reactions below.

File: interfaces/cython/cantera/reaction.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reaction.pyx:1452]: Same as before about math formatting.

File: interfaces/cython/cantera/reactionpath.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reactionpath.pyx]:
```suggestion
        according the the net reaction rates determined by the `Kinetics` object
```

File: interfaces/cython/cantera/reactor.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reactor.pyx]: Why is "relative" in parentheses here? Is it part of the sentence or not? Can we take it out of the parentheses and clarify, relative to what?

File: interfaces/cython/cantera/reactor.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reactor.pyx]:
```suggestion
        a reactor state variable, for example ``'int_energy'``, ``'temperature'``, depending on
        the reactor's equations.
```

File: interfaces/cython/cantera/reactor.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reactor.pyx]:
```suggestion
        towards time ``t`` in seconds, taking as many integrator time steps as necessary.
```

File: interfaces/cython/cantera/reactor.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reactor.pyx]:
```suggestion
        ``reactor`` within the global state vector. That is, this determines the
        absolute index of the component, where ``reactor`` is the index of the
```

File: interfaces/cython/cantera/reactor.pyx
Comment (bryanwweber) [inline interfaces/cython/cantera/reactor.pyx]:
```suggestion
        name of a reactor state variable, for example, ``'int_energy'``, ``'temperature'``, etc.
```

**Review by bryanwweber — DISMISSED** [review]
Phew! Thanks for taking this on @speth. Detail orientation is my jam 😄 As such, I've left 44 comments for further improvements that I noted here! You're welcome 😝 

I'll approve it though because they're all minor and you can merge this after updating them without me needing to re-approve.

File: interfaces/cython/cantera/reaction.pyx
Comment (speth) [reply interfaces/cython/cantera/reaction.pyx:200]: Changing the equation to "E_a" instead of just "E" would make sense, but I think formatting these as code makes sense for the correspondence with the constructor arguments.

File: interfaces/cython/cantera/reaction.pyx
Comment (speth) [reply interfaces/cython/cantera/reaction.pyx:1205]: It wasn't doing it for me when I was testing locally. I don't know how to get Sphinx to link directly to member variables/functions of a different class (and in this case, I think it would have trouble knowing the type of `Reaction.rate` to be able to figure out what class the `rates` property was a member of).

File: interfaces/cython/cantera/composite.py
Comment (speth) [reply interfaces/cython/cantera/composite.py]: Honestly, I don't know. How do you think the name of a different Python package should be formatted? I don't think we have any sort of consistent style. I see cases that are plain text with no formatting, single quotes, and backticks, and there might be others as well.

File: interfaces/cython/cantera/onedim.pyx
Comment (speth) [reply interfaces/cython/cantera/onedim.pyx]: It is the float 0. I changed this to 0.0 with no quotes.

File: interfaces/cython/cantera/onedim.pyx
Comment (speth) [reply interfaces/cython/cantera/onedim.pyx]: How about the following:
>  Derived classes call this function with the flow domain as the ``domain`` and a list of essential non-thermodynamic solution components of the configuration as the ``components``. A different ``domain`` (for example, ``inlet`` or ``outlet``) can be specified either by name or as the corresponding `Domain1D` object itself.

Comment (bryanwweber) [reply interfaces/cython/cantera/onedim.pyx]: Looks good to me

File: interfaces/cython/cantera/reaction.pyx
Comment (bryanwweber) [reply interfaces/cython/cantera/reaction.pyx:1205]: That's fine then ☺️

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [reply interfaces/cython/cantera/composite.py]: I think package names should probably be regular text, unless they're referring to a function or some such. Do the two method names refer to our functions or h5py functions?

Comment (ischoegl) [reply interfaces/cython/cantera/composite.py]: Fwiw, I think it’s helpful to format package names different than other text. Perhaps italics?

File: interfaces/cython/cantera/reaction.pyx
Comment (bryanwweber) [reply interfaces/cython/cantera/reaction.pyx:200]: That's fine, but then we should be consistent in all the reaction-type docstrings. The `TwoTempPlasmaRate` uses math to show these parameters.

File: interfaces/cython/cantera/composite.py
Comment (speth) [reply interfaces/cython/cantera/composite.py]: `write_hdf` and `read_hdf` are methods of the `SolutionArray` class, so single-backticks is correct and links to the corresponding method documentation.

Regarding the package names, I think using regular text works fine when it's written as a proper noun like Numpy, but I find it a little confusing for packages that prefer all lowercase formatting and then you're just talking about pandas in the middle of a sentence. Italics seems like a reasonable choice that wouldn't look out of place in either case.

File: interfaces/cython/cantera/reaction.pyx
Comment (speth) [reply interfaces/cython/cantera/reaction.pyx:200]: Can I suggest that the docstring for `TwoTempPlasmaRate` is out-of-scope for this PR? There are other updates needed there that I would like @BangShiuh to address anyway.

File: interfaces/cython/cantera/composite.py
Comment (bryanwweber) [reply interfaces/cython/cantera/composite.py]: Sounds good to me!

File: interfaces/cython/cantera/reaction.pyx
Comment (bryanwweber) [reply interfaces/cython/cantera/reaction.pyx:200]: Sounds good to me!

**Review by ischoegl — APPROVED** [review]

**Comment by bryanwweber** [comment]
Thanks @speth and @ischoegl! :shipit: In it goes!

**Pull request merged**
